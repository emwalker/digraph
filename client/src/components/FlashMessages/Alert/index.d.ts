import React, { MouseEventHandler, type ReactNode } from 'react';
import { AlertMessageType } from 'components/types';
export type OnCloseType = MouseEventHandler<HTMLButtonElement>;
type Props = {
    children?: ReactNode;
    alert: AlertMessageType;
    onClose?: OnCloseType;
};
export default function Alert({ children, alert, onClose }: Props): React.JSX.Element;
export {};
