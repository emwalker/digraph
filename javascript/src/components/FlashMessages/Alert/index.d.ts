import { Component } from 'react';
import { AlertType } from 'components/types';
declare type Props = {
    message: {
        text: string;
        type: AlertType;
    };
    onClose: () => void;
};
declare class Alert extends Component<Props> {
    get className(): string;
    alertClass: (type: AlertType) => string;
    render: () => JSX.Element;
}
export default Alert;
