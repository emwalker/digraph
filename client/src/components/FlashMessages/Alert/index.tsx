import React, { MouseEventHandler, type ReactNode } from 'react'
import { GoX } from 'react-icons/go'

import { AlertMessageType } from 'components/types'

// https://medium.com/@veelenga/displaying-rails-flash-messages-with-react-5f82982f241c

export type OnCloseType = MouseEventHandler<HTMLButtonElement>

type Props = {
  children?: ReactNode,
  alert: AlertMessageType,
  onClose?: OnCloseType,
}

const alertCasses = {
  ERROR: 'flash-error',
  WARN: 'flash-warn',
  SUCCESS: 'flash-success',
  '%future added value': 'flash-error',
}

export default function Alert({ children, alert, onClose }: Props) {
  const alertClass = alertCasses[alert.type] || 'flash-success'

  const defaultOnClose = () => {
    const removeAlert = window.flashMessages?.removeAlert
    if (!removeAlert) return
    removeAlert(alert.id)
  }

  return (
    <div className={`flash fade in mt-3 mb-3 ${alertClass}`}>
      <button
        className="flash-close"
        onClick={onClose || defaultOnClose}
        type="button"
      >
        <GoX />
      </button>
      { alert.text }
      { children }
    </div>
  )
}
