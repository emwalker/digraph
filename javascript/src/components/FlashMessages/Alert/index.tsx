import React, { Component } from 'react'
import classNames from 'classnames'
import { GoX } from 'react-icons/go'

import { AlertType } from 'components/types'

// https://medium.com/@veelenga/displaying-rails-flash-messages-with-react-5f82982f241c

type Props = {
  message: {
    text: string,
    type: AlertType,
  },
  onClose: () => void,
}

class Alert extends Component<Props> {
  get className(): string {
    return classNames(
      'flash fade in mt-3 mb-3',
      this.alertClass(this.props.message.type),
    )
  }

  alertClass = (type: AlertType) => {
    const classes = {
      ERROR: 'flash-error',
      WARN: 'flash-warn',
      SUCCESS: 'flash-success',
      '%future added value': 'flash-error',
    }
    return classes[type] || 'flash-success'
  }

  render = () => (
    <div className={this.className}>
      <button
        className="flash-close"
        onClick={this.props.onClose}
        type="button"
      >
        <GoX />
      </button>
      { this.props.message.text }
    </div>
  )
}

export default Alert
