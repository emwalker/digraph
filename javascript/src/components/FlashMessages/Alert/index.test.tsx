import React from 'react'
import { shallow } from 'enzyme'

import Alert from './index'

const props = {
  message: 'Gnusto',
}

describe('<Alert />', () => {
  // @ts-expect-error
  const wrapper = shallow(<Alert {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
