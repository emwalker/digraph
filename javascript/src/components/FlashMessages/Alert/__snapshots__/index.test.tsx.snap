// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders 1`] = `
<div
  className="flash fade in mt-3 mb-3 flash-success"
>
  <button
    className="flash-close"
    type="button"
  >
    <GoX />
  </button>
</div>
`;
